test/modal-test.js

```diff
@@ -0,0 +1,163 @@
+  });
+
+  it('Should have warning class', function() {
+    const component = shallow(<Modal onClose={noop} type='warning'/>);
+    expect(component.find('div.modal-header').hasClass('bg-warning')).to.equal(true);
+  });
+
+  it('Modal size should be extra small', function() {
+    const component = shallow(<Modal onClose={noop} size='xs'/>);
+    expect(component.find('div.modal').hasClass('xsmall-panel')).to.equal(true);
+  });
+
+  it('Modal size should be extra small', function() {
+    const component = shallow(<Modal onClose={noop} size='extra-small'/>);
+    expect(component.find('div.modal').hasClass('xsmall-panel')).to.equal(true);
+  });
+
+  it('Modal size should be small', function() {
+    const component = shallow(<Modal onClose={noop} size='sm'/>);
+    expect(component.find('div.modal').hasClass('small-panel')).to.equal(true);
+  });
+
+  it('Modal size should be small', function() {
+    const component = shallow(<Modal onClose={noop} size='small'/>);
+    expect(component.find('div.modal').hasClass('small-panel')).to.equal(true);
+  });
+
+  it('Modal size should be medium', function() {
+    const component = shallow(<Modal onClose={noop} size='md'/>);
+    expect(component.find('div.modal').hasClass('medium-panel')).to.equal(true);
+  });
+
+  it('Modal size should be medium', function() {
+    const component = shallow(<Modal onClose={noop} size='medium'/>);
+    expect(component.find('div.modal').hasClass('medium-panel')).to.equal(true);
+  });
+
+  it('Modal size should be large', function() {
+    const component = shallow(<Modal onClose={noop} size='lg'/>);
+    expect(component.find('div.modal').hasClass('large-panel')).to.equal(true);
+  });
+
+  it('Modal size should be large', function() {
+    const component = shallow(<Modal onClose={noop} size='large'/>);
+    expect(component.find('div.modal').hasClass('large-panel')).to.equal(true);
+  });
+
+  it('Modal size should have a default of empty string', function() {
+    const component = shallow(<Modal onClose={noop} size='auto'/>);
+    expect(component.find('div.modal').hasClass('')).to.equal(true);
+  });
+
+  it('Modal size should have a default of empty string', function() {
+    const component = shallow(<Modal onClose={noop} size='none'/>);
+    expect(component.find('div.modal').hasClass('')).to.equal(true);
+  });
+
+  it('Modal has an icon', function() {
+    const component = shallow(<Modal onClose={noop} icon='fa-fa-plus' />);
+    expect(component.find('i').hasClass('fa-fa-plus')).to.equal(true);
+  });
+
+  it('Modal should have title', function() {
+    const component = shallow(<Modal onClose={noop}  title='Testing title'/>);
+    expect(component.find('div.modal-header').text()).to.contain('Testing title');
+  });
+
+  it('Modal does not have a title', function() {
+    const component = shallow(<Modal onClose={noop}  size='none'/>);
+    expect(component.find('div.modal-header').hasClass('Testing title')).to.equal(false);
```

Comment:
The `modal-header` wouldn't have the class added if there were a title - this would be better written as: `expect(component.find('div.modal-header').text()).to.equal(' ');`

I believe the space is necessary because there is a space between the icon and the title in the header.